net/packet/packet.go

```diff
@@ -55,27 +55,44 @@ func (p *Packet) Pack(w io.Writer, threshold int) error {
 func (p *Packet) packWithoutCompression(w io.Writer) error {
 	buffer := bufPool.Get().(*bytes.Buffer)
 	defer bufPool.Put(buffer)
-	buffer.Reset()
-	n, err := VarInt(p.ID).WriteTo(buffer)
-	if err != nil {
-		panic(err)
-	}
-	// Length
-	_, err = VarInt(int(n) + len(p.Data)).WriteTo(w)
-	if err != nil {
-		return err
+
+	if buffer.Len() < 3 {
+		var padding [3]byte
```

Comment:
This is equivalent to `make([]byte, 3)`, which is wired to have a length of 3 when you are going to write entire p.Data later